### PR TITLE
branchs

### PR DESCRIPTION
**Pull request by lisajako**
branchs

### CODE REVIEW

**Review by lisajako — COMMENTED** [review]
branchs

File: src/data.json [hunk 1]
```diff
@@ -42,6 +42,210 @@
...
+							"multiplier": "need adjustment"
+						]
+					}
+				],
+				"name": "Rocuronium 10mg/ml",
+				"doses": [
+					{
+						"info": "1mg/kg/dose IV",
+						"mg_mcg_mmol":{
+							"multiplier": [1],
```
Comment (bader-nasser) [inline src/data.json:228]: No need for [] if just one number

File: src/data.json [hunk 2]
```diff
@@ -94,7 +381,12 @@
 					"multiplier": [10, 40],
 					"unit": "mcg/hr"
 				},
-				"formula_50ml": "",
+				"formula_50ml": {
+					"full strength": "prepare solution of (_amount_) mg (_amount_/10) ml with (50-(_amount_/10)) ml \n (20mcg/Kg/ml)",
+					"double strength": "prepare solution of (2_amount_) mg (2_amount_/10) ml with (50-(2_amount_/10)) ml \n (40mcg/Kg/ml)",
+					"half strength": "prepare solution of (0.5_amount_) mg (0.5_amount_/10) ml with (50-(0.5_amount/10)) ml \n (10mcg/Kg/ml)",
```
Comment (bader-nasser) [inline src/data.json:387]: No spaces in keys: "full_strength", "double_str." && ...

Comment (bader-nasser) [reply src/data.json:387]: 2_amount is wrong but will fix later

File: src/data.json [hunk 3]
```diff
@@ -113,6 +405,202 @@
...
+			{
+				"name": "midazolam (5mg/ml)",
+				"dose": {
+					"info": "1-4 mcg/kg/min",
+					"multiplier": [1, 4],
+					"unit": "mcg/hr"
+				},
+				"formula_50ml": {
+					"single strength":"prepare solution of (50mg) 10ml of midazolam with 40 ml of 0.9% NS \n (1mg/ml)",
+					"double strength": "prepare solution of (100mg) 20 ml of midazolam with 30 ml of 0,9%NS \n (2mg/ml)"
```
Comment (bader-nasser) [inline src/data.json:475]: no spaces: single_str...

File: src/data.json [hunk 4]
```diff
@@ -113,6 +405,202 @@
...
+				],
+				"incompatible": [
+					"No information"
+				]
+			},
+			{
+				"name": "rocuronium (50mg/ml)",
+				"dose": {
+					"info": "300mcg/kg/hr",
+					"multiplier": [300],
```
Comment (bader-nasser) [inline src/data.json:531]: no []

File: src/data.json [hunk 5]
```diff
@@ -113,6 +405,202 @@
...
+					"unit": "mcg/hr"
+				},
+				"formula_50ml": {
+					"single strength":"prepare solution of (50mg) 5ml of rocuronium with 45 ml of 0.9% NS \n (1mg/ml)",
+					"maximum strength": "preare solution of (250mg) 25ml of rocuronium with25ml of 0.9% NS\n (5mg/ml)"
+				},
```
Comment (bader-nasser) [inline src/data.json:537]: no spaces

File: src/data.json [hunk 6]
```diff
@@ -113,6 +405,202 @@
...
+			{
+				"name": "cisatracurium (2mg/ml)",
+				"dose": {
+					"info": "60-180 mcg/kg/min",
+					"multiplier": [60, 180],
+					"unit": "mcg/hr"
+				},
+				"formula_50ml": {
+					"single strength": "prepare solution of (50mg) 25ml of cisatracurium with 25 ml of 0.9% NS \n (1mcg/ml)",
+					"double strength": "undiluted \n (2mg/ml)"
```
Comment (bader-nasser) [inline src/data.json:559]: spaces

File: src/data.json [hunk 7]
```diff
@@ -113,6 +405,202 @@
...
+			{
+				"name": "ketamine (50mg/ml)",
+				"dose": {
+					"info": "8-45 mcg/kg/min",
+					"multiplier": [8, 45],
+					"unit": "mcg/min"
+				},
+				"formula_50ml": {
+					"single strength": "prepare solution of (50mg) 1ml of ketamine with 49 ml of 0.9% NS \n (1mg/ml)",
+					"double strength": "prepare solution of (50mcg) 1ml of ketamine with 24 ml of 0,9%NS \n (2mg/ml)"
```
Comment (bader-nasser) [inline src/data.json:588]: spaces

File: src/data.json [hunk 8]
```diff
@@ -42,6 +42,210 @@
...
+						"mg_mcg_mmol": {
+						},
+						"ml": {
+							"info": "D50%",
+							"multiplier": [1, 2],
+							"info": "D25%",
+							"multiplier": [2, 4],
+							"info": "D10%", 
+							"multiplier": [5, 10]
+						}
```
Comment (bader-nasser) [inline src/data.json:91]: Will break the app, change to: 
```
"ml": [
  {
	"info": "D50%",
	"multiplier": [1, 2]
  },
  {
	  "info": "D25%",
	  "multiplier": [2, 4]
  },
  {
	  "info": "D10%", 
	  "multiplier": [5, 10]
  }
]
```
and will need to change to code at: https://github.com/lisajako/squh-picu-med-calculator/blob/branchs/src/pages/index.tsx

File: src/data.json [hunk 9]
```diff
@@ -42,6 +42,210 @@
...
 					}
 				]
+			},
+			{
+				"name": "calcium gluconate 10% (100mg/ml, (0.2mmol/ml)",
+				"doses": [
+					{
+						"info": "60-100mg/kg/dose (max. 100mg/dose) IV/IO (0.1-0.2 ml/kg) slow push",
+						"mg_mcg_mmol": {
+							"multiplier": [60, 100], "max 3000"
```
Comment (bader-nasser) [inline src/data.json:52]: "max": 3000,

File: src/data.json [hunk 10]
```diff
@@ -42,6 +42,210 @@
...
+				]
+
+			},
+			{
+				"name": "calcium chloride 10% (100mg/ml), (1.4mmol/ml)", 
+				"doses": [
+					{
+						"info": "10-20mg/kg/dose (max 100mg/dose) IV/IO (0.1-0.2 ml/kg) Slow Push via central line, dilute if peripheral line",
+						"mg_mcg_mmol": {
+							"multiplier": [10, 20], "max 100",
```
Comment (bader-nasser) [inline src/data.json:68]: max

File: src/data.json [hunk 11]
```diff
@@ -42,6 +42,210 @@
...
+				]
+			},
+			{
+				"name": "Adenosine (6mg/2ml)",
+				"doses": [
+					{
+						"info": "0.1mg/kg IV/IO rapid bolus in central line (max 6mg); Second dose: 0.2 mg/kg (max 12mg)",
+						"mg_mcg_mmol": {
+							"multiplier": [0.1], "max 6",
+							"multiplier": [0.2], "max 12",
```
Comment (bader-nasser) [inline src/data.json:102]: max in many places

Comment (bader-nasser) [reply src/data.json:102]: No need for [] & it'll break the app if added for a single value

can use same key twice, change to: 
```json
"mg_mcg_mmol": {
  "unit": "mg",
  "doses": [
    {
      "multiplier": 0.1, 
      "max": 6
    },
    {
      "multiplier": 0.2, 
      "max": 12
    }
  ]
}
```

File: src/data.json [hunk 12]
```diff
@@ -42,6 +42,213 @@
...
+							},
+							{
+								"info": "D25%",
+								"multiplier": [2, 4]
+							},
+							{
+								"info": "D10%", 
+								"multiplier": [5, 10]
+							}
+						  ],
```
Comment (bader-nasser) [inline src/data.json:97]: no ,

File: src/data.json [hunk 13]
```diff
@@ -42,6 +42,213 @@
...
+						"mg_mcg_mmol":{
+							"multiplier": [0.1, 2]
+							},
+						"ml": "need adjustment"
+						
+					}
+				]
+			}
+		],
+		"pediatric Intubation Medications": [
```
Comment (bader-nasser) [inline src/data.json:141]: small letters and no spaces

File: src/data.json [hunk 14]
```diff
@@ -42,6 +42,213 @@
...
+				],
+				"name": "Ketamine (50mg/ml), (Prepare a solution of 50mg ketamine with 9ml 0.9% NS (5mg/ml))", 
+				"doses": [
+					{
+						"info": "1-2 mg/kg/dose IV",
+						"mg_mcg_mmol": {
+							"multiplier": [1, 2], 
+							"unit": "mg"},
+						"ml": [
+							"multiplier": "need adjustment"
```
Comment (bader-nasser) [inline src/data.json:184]: add `]` at the end

File: src/data.json [hunk 15]
```diff
@@ -71,11 +303,69 @@
...
 				"incompatible": [
 					"sodium bicarbonate",
-					"**edit this and add more**"
+					"aminophylline",
+					"omeprazole",
+					"heparine"
+				]
+			},
+			{
+				"dobutamine (5mg/ml)",
```
Comment (bader-nasser) [inline src/data.json:321]: "name": "dob...",

File: src/data.json [hunk 16]
```diff
@@ -71,11 +303,69 @@
...
+				"dose": "5-20 mcg/kg/min",
+				"formula_50ml":{
+					"text": "Prepare solution of (_amount_) ml Dobutamine in (_ns_amount_) ml of 0.9% NS \n (1ml/hr = 1 mgc/kg/min)",
+					"multiplier": 30,
+					"ns_multiplier": "adjust"
```
Comment (bader-nasser) [inline src/data.json:326]: `ns_multiplier` is number multiplied by the weight, no need to add it if it's the same as the med's `multiplier`
The 50 is added in the code: https://github.com/lisajako/squh-picu-med-calculator/blob/branchs/src/components/inotropic-infusions.tsx#L54
I may add a divider in other categories

File: src/data.json [hunk 17]
```diff
@@ -71,11 +303,69 @@
...
+					"aminophylline",
+					"omeprazole",
+					"heparine"
+				]
+			},
+			{
+				"name": "milrinone (1mg/ml)",
+				"dose": "0.25-0.75 mcg/kg/min",
+				"formula_50ml":{
+					"text":"Prepare solution of (10mg) 10 ml milrinone in 40 ml of 0.9% NS or D5W \n (200mcg/ml)"
```
Comment (bader-nasser) [inline src/data.json:350]: if no multiplier or any calculations, I add the text directly: 
```json
"formula_50ml": "Prepare solution of (10mg) 10 ml milrinone in 40 ml of 0.9% NS or D5W \n (200mcg/ml)"
```
similar to what I did with Propofol

File: src/data.json [hunk 18]
```diff
@@ -5,14 +5,14 @@
 				"name": "Epinephrine 1:10000 (1mg/10ml)",
 				"doses": [
 					{
-						"info": "0.01 mg/kg/dose (max 1 mg) IV/IO (0.1 ml/kg of 1:10000)",
+						"info": "0.01 mg/kg/dose max 1 mg) IV/IO (0.1 ml/kg of 1:10000)",
```
Comment (bader-nasser) [inline src/data.json:8]: (max

File: src/data.json [hunk 19]
```diff
@@ -42,6 +42,213 @@
...
+							},
+						"ml": "need adjustment"
+						
+					}
+				]
+			}
+		],
+		"pediatric Intubation Medications": [
+			{
+			    "name": "Atropine (0.5mg/ml), (Prepare a solution of 0.5mg of atropine with 4ml of 0.9%NS (100mcg/ml))",
```
Comment (bader-nasser) [inline src/data.json:143]: Why added "prepare a sol..". to the name?  shouldn't this be shown in `ml` column or what?

File: src/data.json (hunk 12 above)
Comment (bader-nasser) [reply src/data.json:97]: add a new line with these
```
    }
  ]
},
```

File: src/data.json [hunk 20]
```diff
@@ -55,10 +307,35 @@
 				"compatible": [
 					"Dextrose 5-10%",
 					"sodium chloride 0.9%",
-					"**edit this and add more**"
+					"**add more**"
```
Comment (bader-nasser) [inline src/data.json:310]: add the rest

File: src/data.json (hunk 19 above)
Comment (lisajako) [reply src/data.json:159]: to standradize administration od boluses, its required but thought better to be added with the name.

File: src/data.json [hunk 21]
```diff
@@ -42,6 +42,121 @@
...
+							"max": 1
+						}
+					}
+				]
+			},
+			{
+				"name": "dextrose",
+				"doses": [
+					{
+						"info": " IV/IO: 1-2 mL/kg D50%, via central line only, 2-4 mL/kg D25%, via central line only, 5-10 mL/kg D10%",
```
Comment (bader-nasser) [inline src/data.json:84]: add `\n` before the text if you want to show it in a new line
```json
"info": "IV/IO:\n 1-2ml... only, \n 2-4..."
```

File: src/data.json [hunk 22]
```diff
@@ -42,6 +42,121 @@
...
+						"mg_mcg_mmol": "",
+						"ml": [
+							{
+								"info": "D50%",
```
Comment (bader-nasser) [inline src/data.json:88]: maybe it's better to call it `label` instead of `info`, but still not sure about how to handle these changing data!

**Comment by bader-nasser** [comment]
Will merge and fix any issue later (hopefully none)

**Pull request merged**